[NEUT-599] [WIP] Migrate Content Container widget to Backpack (#2527)
Skyscanner/backpack: packages/bpk-component-content-cards/README.md

# Content Cards

> Backpack example component.

## Installation

```sh
npm install bpk-component-content-cards --save-dev
```

## Usage

```js
import React from 'react';
import BpkContentCards from 'bpk-component-code';

export default () => <BpkContentCards />;
```

## Props

| Property  | PropType | Required | Default Value |
| --------- | -------- | -------- | ------------- |

> | className | string   | false    | null          |

Reviewer: Looking at the examples you have created this table doesn't look to match up

# BpkContentCards

> > Backpack example component.

Author: @olliecurtis will [this URL]( https://backpack.github.io/storybook-prs/2527) still be active after this PR has been merged and deleted? If so I can link to that in the README. As for the docs link, how is the link decided? Is it automated or can we set it? Would be good if we can already decide on what the URL will be so we can link to it already.

> > Backpack example component.

Reviewer: >@olliecurtis will this URL still be active after this PR has been merged and deleted?

Yes it will be however once it is merged you can use the main deployed version once its been merged of https://backpack.github.io/storybook

>As for the docs link, how is the link decided? Is it automated or can we set it? Would be good if we can already decide on what the URL will be so we can link to it already.

This gets generated when the docs are created so I imagine it would end up being backpack.github.io/components/content-cards

> > Backpack example component.

Reviewer 2: missing description of the component here, or just remove it
```suggestion
```

## Installation

```sh

> npm install bpk-component-content-cards --save-dev

Reviewer: I would for now keep it as the single one as once we move to only single package we will update it then

```

## Usage

> ```js

Reviewer 3: ```suggestion
```ts
```

import React from 'react';
import BpkContentCards from 'bpk-component-content-cards';

<BpkContentCards
    headerText='Header'
    cards=[
        {

>         imageLink: 'http://skyscanner.net/card-1-image-link.jpg',
>         imageAlt: '',
>         headline: 'Card headline',
>         description: 'Card description',
>         ctaLink: 'http://skyscanner.net/card-1-cta-link.html',

Reviewer 2: Just thinking out loud, maybe this API could be improved, now that we decouple it from the context of Sponsored Destination Explorer. E.g. `imageLink` can be interpreted as "where does the image link to" and not "image to render src/href". In a similar way, `ctaLink` might be something different, as it suggests there's a Call To Action (CTA), but actually whole card is a link to some content.

Examples (just some quick thought):
```suggestion
        image: { url: 'http://skyscanner.net/card-1-image-link.jpg', alt: 'White sand beach in a shiny weather of Australian coast', },
        headline: 'Card headline',
        description: 'Card description',
        link: 'http://skyscanner.net/example.html', // or href/url/keep ctaLink
```

I am fine with current API, just thinking if we can improve it right now.

Author: <img width="817" alt="Screenshot 2022-08-11 at 17 09 58" src="https://user-images.githubusercontent.com/58072716/184166851-fe3c3450-12c0-4e19-83ff-d5e3e8969f01.png">
this creates a problem with the eslint as it thinks that we're passing the word 'image' as the alt and disabling the check for that line doesn't work

Author: I changed it from image to img and it fixes the problem

Reviewer 2: you don't have to make any changes here right now, I'm raising it to open a discussion, maybe we can involve other people in this

# BpkContentCards

> Backpack example component.

- [Storybook link](https://backpack.github.io/storybook)
- [Content Cards Docs](backpack.github.io/components/content-cards)

## Installation

```sh
npm install bpk-component-content-cards --save-dev
```

## Usage

```tsx
import React from 'react';
import BpkContentCards from '@skyscanner/backpack-web/bpk-component-content-cards';

<BpkContentCards

>     headerText='Header'

Reviewer 2: sorry 😅 

Will it make sense to change to `headingText` or just `heading`? `headerText` seems to be a bit off, but maybe it's my head that has weird image of what 'headers' are

Author: Changed to `heading` 👍
